File: app/src/main/java/com/tomtom/timetoleave/CustomSnackbar.java
package com.tomtom.timetoleave;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.core.view.ViewCompat;

import com.google.android.material.snackbar.BaseTransientBottomBar;

public class CustomSnackbar extends BaseTransientBottomBar<CustomSnackbar> {
    private CustomSnackbar(ViewGroup parent, View content,
                           ContentViewCallback contentViewCallback) {
        super(parent, content, contentViewCallback);
    }

    private static class ContentViewCallback implements
            BaseTransientBottomBar.ContentViewCallback {

        private final View content;

        private ContentViewCallback(View content) {
            this.content = content;
        }

        @Override
        public void animateContentIn(int delay, int duration) {
            content.setAlpha(0f);
            ViewCompat.animate(content)
                    .alpha(1f).setDuration(duration)
                    .setStartDelay(delay);
        }

        @Override
        public void animateContentOut(int delay, int duration) {
            content.setAlpha(1f);
            ViewCompat.animate(content)
                    .alpha(0f)
                    .setDuration(duration)
                    .setStartDelay(delay);
        }
    }

    public static CustomSnackbar make(ViewGroup parent, int duration, int layout) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View content = inflater.inflate(layout, parent, false);

        ContentViewCallback callback= new ContentViewCallback(content);
        CustomSnackbar customSnackbar = new CustomSnackbar(parent, content, callback);

        customSnackbar.setDuration(duration);
        return customSnackbar;
    }

    public void setText(CharSequence text) {
        TextView textView = getView().findViewById(R.id.text_view_dialog_recalculation_info);
        textView.setText(text);
    }

    public void setAction(CharSequence text, final View.OnClickListener listener) {
        Button actionView = getView().findViewById(R.id.button_dialog_recalculation_ok);
        actionView.setText(text);
        actionView.setVisibility(View.VISIBLE);
        actionView.setOnClickListener(listener);
    }
}

File: TimeToLeave-ktx/src/main/java/com/tomtom/timetoleave/CustomSnackbar.kt
package com.tomtom.timetoleave

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView

import androidx.core.view.ViewCompat

import com.google.android.material.snackbar.BaseTransientBottomBar

class CustomSnackbar constructor(parent: ViewGroup, content: View, contentViewCallback: ContentViewCallback) :
        BaseTransientBottomBar<CustomSnackbar>(parent, content, contentViewCallback) {

    class ContentViewCallback constructor(private val content: View) : com.google.android.material.snackbar.ContentViewCallback {

        override fun animateContentIn(delay: Int, duration: Int) {
            content.alpha = 0f
            ViewCompat.animate(content)
                    .alpha(1f).setDuration(duration.toLong()).startDelay = delay.toLong()
        }

        override fun animateContentOut(delay: Int, duration: Int) {
            content.alpha = 1f
            ViewCompat.animate(content)
                    .alpha(0f)
                    .setDuration(duration.toLong()).startDelay = delay.toLong()
        }
    }

    fun setText(text: CharSequence) {
        val textView = getView().findViewById<TextView>(R.id.text_view_dialog_recalculation_info)
        textView.text = text
    }

    fun setAction(text: CharSequence, listener: (Any) -> Unit) {
        getView().findViewById<Button>(R.id.button_dialog_recalculation_ok).apply {
            this.text = text
            this.visibility = View.VISIBLE
            this.setOnClickListener(listener)
        }
    }

    companion object {
        fun make(parent: ViewGroup, duration: Int, layout: Int): CustomSnackbar {
            val inflater = LayoutInflater.from(parent.context)
            val content = inflater.inflate(layout, parent, false)

            val callback = ContentViewCallback(content)
            return CustomSnackbar(parent, content, callback).apply {
                this.duration = duration
            }
        }
    }
}
